memorious/operations/db.py

```diff
@@ -4,17 +4,49 @@
 from memorious.core import datastore
 
 
-def db(context, data):
-    """Insert or update `data` as a row into specified db table"""
-    table = context.params.get("table", context.crawler.name)
+def _recursive_upsert(context, params, data):
```

Comment:
Can we separate the "recursive" concern from the "upsert" concern somehow, make them two functions that call each other?